controllers/helpers.go

```diff
@@ -32,6 +32,81 @@ import (
 	"k8s.io/apimachinery/pkg/util/intstr"
 )
 
+// filtering container meta data containing a list of filtered out containers and the reasoning for the
+// the filter.
+type filteringContMeta struct {
```

Comment:
nit: this name is a little terse, could it just be `filteredContainers`? Also, could `removed` be clear if it's ids or names?

Reply:
Also, I might suggest that you use a slice of this struct, rather than a struct of slices? That way the removed/reasoning tuples are more closely packaged, rather than relying on all users of this struct to make sure the two inner slices have matching orders

Reply:
I've added to the comments that removed is specifically the names of the containers, let me know if there was any other place to clarify that I missed.